modules/http/http.c

```diff
@@ -491,6 +223,324 @@ http_dd_set_timeout(LogDriver *d, glong timeout)
+  "header_out",
+  "data_in",
+  "data_out",
+  "ssl_data_in",
+  "ssl_data_out",
+};
+
+static gint
+_curl_debug_function(CURL *handle, curl_infotype type,
+                     char *data, size_t size,
+                     void *userp)
+{
+  if (!G_UNLIKELY(debug_flag))
+    return 0;
+
+  g_assert(type < sizeof(curl_infotype_to_text)/sizeof(curl_infotype_to_text[0]));
+
+  const gchar *text = curl_infotype_to_text[type];
+  gchar *sanitized = _sanitize_curl_debug_message(data, size);
+  msg_debug("cURL debug",
+            evt_tag_str("type", text),
+            evt_tag_str("data", sanitized));
+  g_free(sanitized);
+  return 0;
+}
+
+static size_t
+_curl_write_function(char *ptr, size_t size, size_t nmemb, void *userdata)
+{
+  // Discard response content
+  return nmemb * size;
+}
+
+static const gchar *
+_format_persist_name(const LogPipe *s)
+{
+  const HTTPDestinationDriver *self = (const HTTPDestinationDriver *)s;
+  static gchar persist_name[1024];
+
+  if (s->persist_name)
+    g_snprintf(persist_name, sizeof(persist_name), "http.%s", s->persist_name);
+  else
+    g_snprintf(persist_name, sizeof(persist_name), "http(%s,)", self->url);
+
+  return persist_name;
+}
+
+static const gchar *
+_format_stats_instance(LogThreadedDestDriver *s)
+{
+  static gchar stats[1024];
+
+  HTTPDestinationDriver *self = (HTTPDestinationDriver *) s;
+
+  g_snprintf(stats, sizeof(stats), "http,%s", self->url);
+
+  return stats;
+}
+
+
+/* Set up options that are static over the course of a single configuration,
+ * request specific options will be set separately
+ */
+static void
+_setup_static_options_in_curl(HTTPDestinationDriver *self)
+{
+  curl_easy_reset(self->curl);
+
+  curl_easy_setopt(self->curl, CURLOPT_WRITEFUNCTION, _curl_write_function);
+
+  curl_easy_setopt(self->curl, CURLOPT_URL, self->url);
+
+  if (self->user)
+    curl_easy_setopt(self->curl, CURLOPT_USERNAME, self->user);
+
+  if (self->password)
+    curl_easy_setopt(self->curl, CURLOPT_PASSWORD, self->password);
+
+  if (self->user_agent)
+    curl_easy_setopt(self->curl, CURLOPT_USERAGENT, self->user_agent);
+
+  if (self->ca_dir)
+    curl_easy_setopt(self->curl, CURLOPT_CAPATH, self->ca_dir);
+
+  if (self->ca_file)
+    curl_easy_setopt(self->curl, CURLOPT_CAINFO, self->ca_file);
+
+  if (self->cert_file)
+    curl_easy_setopt(self->curl, CURLOPT_SSLCERT, self->cert_file);
+
+  if (self->key_file)
+    curl_easy_setopt(self->curl, CURLOPT_SSLKEY, self->key_file);
+
+  if (self->ciphers)
+    curl_easy_setopt(self->curl, CURLOPT_SSL_CIPHER_LIST, self->ciphers);
+
+  curl_easy_setopt(self->curl, CURLOPT_SSLVERSION, self->ssl_version);
+  curl_easy_setopt(self->curl, CURLOPT_SSL_VERIFYHOST, self->peer_verify ? 2L : 0L);
+  curl_easy_setopt(self->curl, CURLOPT_SSL_VERIFYPEER, self->peer_verify ? 1L : 0L);
+
+  curl_easy_setopt(self->curl, CURLOPT_DEBUGFUNCTION, _curl_debug_function);
+  curl_easy_setopt(self->curl, CURLOPT_VERBOSE, (long) (debug_flag || trace_flag));
```

Comment:
Regression.
debug_flag and trace_flag can be changed runtime with syslog-ng-ctl, without reload. With this patch, the http loglevel cannot follow such change. That's why we explicitely set verbose level to one in the original version, and have an early return based on debug_flag in _curl_debug_function.